libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,462 @@
+ * License (a copy is located in LICENSE.txt at the top-level directory).
+ *
+ * Copyright (C) 2018, The University of Texas at Austin. All rights reserved.
+ * UNIVERSITY EXPRESSLY DISCLAIMS ANY AND ALL WARRANTIES CONCERNING THIS
+ * SOFTWARE AND DOCUMENTATION, INCLUDING ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR ANY PARTICULAR PURPOSE, NON-INFRINGEMENT AND WARRANTIES OF
+ * PERFORMANCE, AND ANY WARRANTY THAT MIGHT OTHERWISE ARISE FROM COURSE OF
+ * DEALING OR USAGE OF TRADE.  NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH
+ * RESPECT TO THE USE OF THE SOFTWARE OR DOCUMENTATION. Under no circumstances
+ * shall University be liable for incidental, special, indirect, direct or
+ * consequential damages or loss of profits, interruption of business, or
+ * related expenses which may arise from use of Software or Documentation,
+ * including but not limited to those resulting from defects in Software and/or
+ * Documentation, or loss or inaccuracy of data of any kind.
+ */
+
+#include "katana/analytics/k_shortest_paths/ksssp.h"
+
+#include "katana/AtomicHelpers.h"
+#include "katana/Reduction.h"
+#include "katana/Statistics.h"
+#include "katana/TypedPropertyGraph.h"
+#include "katana/analytics/BfsSsspImplementationBase.h"
+#include "katana/analytics/kSsspImplementationBase.h"
+#include "katana/gstl.h"
+
+using namespace katana::analytics;
+
+struct Path {
+  uint32_t parent;
+  const Path* last{nullptr};
+};
+
+struct NodeCount : public katana::AtomicPODProperty<uint32_t> {};
+
+struct NodeMax : public katana::AtomicPODProperty<uint32_t> {};
+
+using EdgeWeight = katana::UInt32Property;
+
+using NodeData = std::tuple<NodeCount, NodeMax>;
+using EdgeData = std::tuple<EdgeWeight>;
+using Weight = uint32_t;
+using Distance = uint32_t;
+
+constexpr static const bool kTrackWork = false;
+constexpr static const unsigned kChunkSize = 64U;
+
+using PSchunk = katana::PerSocketChunkFIFO<kChunkSize>;
+
+class PathAlloc {
+public:
+  Path* NewPath() {
+    Path* path = allocator_.allocate(1);
+    allocator_.construct(path, Path());
+    return path;
+  }
+
+  void DeletePath(Path* path) {
+    allocator_.destroy(path);
+    allocator_.deallocate(path, 1);
+  }
+
+private:
+  katana::FixedSizeAllocator<Path> allocator_;
+};
+
+template <
+    typename GraphTy, typename Item, typename PushWrap, typename EdgeRange>
+bool
+CheckReachabilityAsync(
+    GraphTy* graph, const typename GraphTy::Node& source,
+    const PushWrap& push_wrap, const EdgeRange& edge_range,
+    uint32_t report_node) {
+  using FIFO = katana::PerSocketChunkFIFO<kChunkSize>;
+  using WL = FIFO;
+
+  using Loop = katana::ForEach;
+
+  using GNode = typename GraphTy::Node;
+
+  Loop loop;
+
+  graph->template GetData<NodeCount>(source) = 1;
+  katana::InsertBag<Item> initBag;
+
+  push_wrap(initBag, source, 1, "parallel");
+
+  loop(
+      katana::iterate(initBag),
+      [&](const Item& item, auto& ctx) {
+        for (auto ii : edge_range(item)) {
```

Comment:
just use the `Edges` function everywhere

Reply:
edge_range is a variable part of the function call (called [here](https://github.com/dqawami/katana/blob/1d88284a0f19755e80553ad453a2971142e1fae2/libgraph/src/analytics/k_shortest_paths/ksssp.cpp#L335)), should I still be replacing it?

Reply:
ok it's fine then